### DISCUSSION TITLE
Segment lengths are continuous/gapless

### DISCUSSION THREAD

**Discussion by p4-k4** [openai/whisper#1036] [General]
Currently, we get continuous/gapless segments where segments are hard "butted up" against each other (indicative by the end-time of the previous and start-time of the proceeding segment (`1 ends at 00:00:09,600 while 2 starts at 00:00:09,600 but in fact, there is a gap between them`).

```srt
1
00:00:00,000 --> 00:00:09,600
New Zealand's first ever event

2
00:00:09,600 --> 00:00:15,720
Wingfoil World Tour event.

3
00:00:15,720 --> 00:00:20,520
Global competitors were welcomed at Whareroa Marae with a special opening ceremony to mark
```

 Not sure if it's work in progress but would love to see segments start/end times matching with speech segment lengths that are true to the respective speech segments.
 
 On the other hand, being able to specify a duration that; if the time between each segment is less than `x`specified time, butt them up against each other. WRT subtitles, this would prevent "flashing" subtitles where the gap between them is so short that we should probably just extend it out to the next one.

**Comment by p4-k4** [comment]
Ending up putting something together that converts Whisper word-level generated SRT's to segment-level SRT's by condensing sentences down to one segment and reassigning start and end timestamps in accordance to first and last words.

When running Whisper with the `--word_timestamps` flag set to `True`, we get this:
```
1
00:00:00,000 --> 00:00:00,500
<u>If</u> you are driving around the Hastings or Napier, you will see plenty of low-vacancy signs.

------------ 2-17 ------------

18
00:00:06,560 --> 00:00:07,160
If you are driving around the Hastings or Napier, you will see plenty of low-vacancy signs.
```

`subseg input.srt output.srt` gets us this (Note the start and end timestamps):
```
1
00:00:00,000 --> 00:00:07,160
If you are driving around the Hastings or Napier, you will see plenty of low-vacancy signs.

```

Install dart, or run compiled in `/bin`.
Clone: `git clone https://github.com/p4-k4/subseg.git`
Repo: https://github.com/p4-k4/subseg

**Comment by ClaireCJS** [comment]
Yeah, I kind of thought that if whisper was outputting an end time, it would be the actual end time of the speech, not... the beginning of the next speech.
